predict.py

```diff
@@ -48,21 +42,22 @@ def setup(self, weights: Optional[Path] = None):
         self.downloader = Downloader()
         self.device = "cuda" if torch.cuda.is_available() else "cpu"
 
-        from src.exllama_predictor import ExllamaWrapper
 
         base_weights = maybe_download_with_pget(
             LOCAL_DEFAULT_INFERENCE_WEIGHTS_PATH,
             REMOTE_DEFAULT_INFERENCE_WEIGHTS_PATH,
             REMOTE_DEFAULT_INFERENCE_FILES_TO_DOWNLOAD,
         )
-        self.exllama_wrapper = ExllamaWrapper(base_weights, fused_attn=USE_FUSED_ATTN)
+
+        self.engine = ENGINE(base_weights, **ENGINE_KWARGS)
 
         if weights is not None and weights.name == "weights":
             # bugfix
             weights = None
         if weights:
             # If weights are passed in, they are LoRa weights
             # so we need to download the fp16 weights and load with peft
+            # TODO: pass engine and weights both in here. 
```

Comment:
curious what your thoughts on what this should look like?

like should it be `self.engine.initialize_peft(weights)`, or some other approach?